oauthproxy.go

```diff
@@ -298,18 +300,18 @@ func NewOAuthProxy(opts *Options, validator func(string) bool) *OAuthProxy {
 		UserInfoPath:      fmt.Sprintf("%s/userinfo", opts.ProxyPrefix),
 
 		ProxyPrefix:          opts.ProxyPrefix,
-		provider:             opts.provider,
+		provider:             opts.GetProvider(),
 		providerNameOverride: opts.ProviderName,
-		sessionStore:         opts.sessionStore,
+		sessionStore:         opts.GetSessionStore(),
 		serveMux:             serveMux,
 		redirectURL:          redirectURL,
 		whitelistDomains:     opts.WhitelistDomains,
 		skipAuthRegex:        opts.SkipAuthRegex,
 		skipAuthPreflight:    opts.SkipAuthPreflight,
 		skipJwtBearerTokens:  opts.SkipJwtBearerTokens,
-		jwtBearerVerifiers:   opts.jwtBearerVerifiers,
-		compiledRegex:        opts.compiledRegex,
-		realClientIPParser:   opts.realClientIPParser,
+		jwtBearerVerifiers:   opts.GetJWTBearerVerifiers(),
+		compiledRegex:        opts.GetCompiledRegex(),
+		realClientIPParser:   opts.GetRealClientIPParser(),
```

Comment:
I've always wondered why `OAuthProxy` doesn't just have an `opts *Options` member (or embed it)...

Reply:
I agree to an extent with this sentiment, it does seem weird to copy everything over. 

One of the reasons for not doing so is that the OAuth2 Proxy doesn't need the whole Options after it has been initiated.
Once the options structure has been broken down I imagine we will probably replace a lot of this with the smaller structs being members of OAuth2 Proxy as appropriate